be1-go/channel/reaction/mod.go

```diff
@@ -25,17 +26,31 @@ const msgID = "msg id"
 const failedToDecodeData = "failed to decode message data: %v"
 
 // NewChannel returns a new initialized reaction channel
-func NewChannel(channelPath string, hub channel.HubFunctionalities, log zerolog.Logger) Channel {
+func NewChannel(channelPath string, hub channel.HubFunctionalities, log zerolog.Logger) *Channel {
```

Comment:
I am not sure to understand why it is *Channel and not channel.Channel ?

Reply:
Because we rely a lot on the fact that NewChannel returned a Channel and if I made the change I would have to cast a lot and it would decrease the readability of the code. We discussed the issue in the last go meeting I believe.

Reply:
Ok perfect